frontend/src/components/node/Details.tsx

```diff
@@ -150,82 +155,108 @@ export default function NodeDetails() {
     );
   }
 
+  function DrainDialog() {
+    function useDrain(drainNode: Node) {
```

Comment:
Just one nit: When we choose `useSMTH`, this looks like a hook and gets me nervous 🙂 
You can just move the 2 functions you have here into the onConfirm arrow func body.